### PR TITLE
Eliminating the need for namespace internal linkage in World class.

### PR DESCRIPTION
**Pull request by ApostolosBouz**
This is a stepping stone edit.

To my understanding, in order to develop features like stored and reusable Entities, Undo/Redo, and easier World editing from UI, we first need to make World class able to be inherited in a meaningful way.

If this PR passes, next step is to derive separate Worlds for Car, Cube, Forest, etc. from a single World class.

### CODE REVIEW

**Comment by PanosK92** [comment]
Thanks for the PR.

What's the benefit of moving things from the anonymous namespace into the class itself?

**Comment by ApostolosBouz** [comment]
Thanks for the reply. I have some reasons in mind, based on Spartan's roadmap, and overall design clarity:

1. Bringing functions and variables into the scope of the class, so that they are part of the class implementation, so that the World can be subclassed effectively.
2. Better design clarity and intend.
3. Better emphasis of class role.
4. Potential performance improvement by inlining the previous anonymous namespace members.
5. Better team collaboration by class split.

To sum up, and having mentioned that this is a "stepping stone" edit:

1. As far as functionality in concerned, nothing has changed yet (maybe a borderline-noticeable performance on world initialization). 
2. My intend is to move forward specializing World class, to separate derived Worlds based on their content to make adding and editing content easier.

**Comment by ApostolosBouz** [comment]
Closing - no actual meassurable value.
